src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -106,7 +107,7 @@ private void processZipFile(CloudBlobContainer container, String zipFilename)
 
     private void delete(CloudBlockBlob cloudBlockBlob, Envelope envelope) {
         try {
-            if (cloudBlockBlob != null) {
+            if (cloudBlockBlob != null && envelope.getStatus() == Status.PROCESSED) {
```

Comment:
Delete method should not be called if envelope hasn' t been processed ?

Reply:
Not at the moment as we have moved from a model where all failures were upload failures to a model with different failures. So for the time being to avoid issues (such as the current build failure in master) we'd better be stricter. Later this can be modified as the file can be deleted earlier, but re-processing partially processed files needs also work.

Reply:
I am failing to understand why we are doing this and that too because of build failure ? I think irrespective of the model we choose if previous step fails this step should not executed at all. I think we need to review this. Happy to discuss this in person or on slack if needed :)